src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/GCHandleExtensions.cs

```diff
@@ -0,0 +1,36 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Runtime.InteropServices
+{
+    /// <summary>
+    /// Provides extension methods to operate with GC handles.
+    /// </summary>
+    public static class GCHandleExtensions
+    {
+        // The following methods are strongly typed generic specifications only on
+        // PinnedGCHandles with correct type.
+
+        /// <summary>
+        /// Retrieves the address of string data in a <see cref="PinnedGCHandle{T}"/> of array.
+        /// </summary>
+        /// <returns>The address of the pinned array.</returns>
+        [CLSCompliant(false)]
+        public static unsafe T* GetAddressOfArrayData<T>(
+#nullable disable // Nullable oblivious because no covariance between PinnedGCHandle<T> and PinnedGCHandle<T?>
+            this PinnedGCHandle<T[]> handle)
+#nullable restore
+            => (T*)handle.GetAddressOfObjectData();
```

Comment:
Should those be implemented separately here to avoid runtime checks GCHandle does for performance?

Reply:
Good point. Yes the check can't be eliminated here.